### PR TITLE
Enable itest on CI for Cass and Ispn backend

### PR DESCRIPTION
**Pull request by lucasponce**
This PRs enable the itest for Cassandra and Ispn backend.
This is separated on purpose as perhaps in the future we may want to stop Cass on .next.
Also, there are several fixes and tweaks from itest execution but itest and persistence test should be working on CI.
This is not fully finished yet, so in a following PR I will add some missing changes but it is good to consolidate it.

### CODE REVIEW

**Comment by lucasponce** [comment]
So, travis seems can handle both loads.
Some info from executions (first cass, next ispn):

```
[INFO] Reactor Summary:
[INFO] 
[INFO] Hawkular Alerting .................................. SUCCESS [  0.076 s]
[INFO] Hawkular Alerting: Api ............................. SUCCESS [  7.108 s]
[INFO] Hawkular Alerting: Commons ......................... SUCCESS [  0.964 s]
[INFO] Hawkular Alerting: Actions parent .................. SUCCESS [  0.005 s]
[INFO] Hawkular Alerting: Actions Api ..................... SUCCESS [  1.355 s]
[INFO] Hawkular Alerting: Filter Cache API ................ SUCCESS [  0.191 s]
[INFO] Hawkular Alerting: Engine Extensions parent ........ SUCCESS [  0.001 s]
[INFO] Hawkular Alerting: Events Aggregation Extension .... SUCCESS [  6.556 s]
[INFO] Hawkular Alerting: Engine .......................... SUCCESS [ 36.872 s]
[INFO] Hawkular Alerting: Actions Implementation .......... SUCCESS [  0.209 s]
[INFO] Hawkular Alerting: Actions Tests ................... SUCCESS [  0.215 s]
[INFO] Hawkular Alerting: Action Plugins .................. SUCCESS [  0.001 s]
[INFO] Hawkular Alerting: Action Email Plugin ............. SUCCESS [ 13.212 s]
[INFO] Hawkular Alerting: Action File Plugin .............. SUCCESS [  2.121 s]
[INFO] Hawkular Alerting: Rest Handlers ................... SUCCESS [  1.684 s]
[INFO] Hawkular Alerting: Integration Tests ............... SUCCESS [05:20 min]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 06:31 min
[INFO] Finished at: 2017-06-28T11:05:36+00:00
[INFO] Final Memory: 77M/671M
[INFO] ------------------------------------------------------------------------

travis_time:end:02449f1e:start=1498647542157383468,finish=1498647937147305167,duration=394989921699
[0K
[32;1mThe command "./mvnw -s .travis.maven.settings.xml verify -Pitest-cass" exited with 0.[0m


[INFO] 
[INFO] --- exec-maven-plugin:1.6.0:exec (stop-alerting-server) @ hawkular-alerts-itests ---
Stopping HawkularServer
[INFO] 
[INFO] --- maven-failsafe-plugin:2.19.1:verify (final-verify) @ hawkular-alerts-itests ---
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] Hawkular Alerting .................................. SUCCESS [  0.078 s]
[INFO] Hawkular Alerting: Api ............................. SUCCESS [  3.113 s]
[INFO] Hawkular Alerting: Commons ......................... SUCCESS [  0.195 s]
[INFO] Hawkular Alerting: Actions parent .................. SUCCESS [  0.001 s]
[INFO] Hawkular Alerting: Actions Api ..................... SUCCESS [  0.895 s]
[INFO] Hawkular Alerting: Filter Cache API ................ SUCCESS [  0.028 s]
[INFO] Hawkular Alerting: Engine Extensions parent ........ SUCCESS [  0.001 s]
[INFO] Hawkular Alerting: Events Aggregation Extension .... SUCCESS [  5.185 s]
[INFO] Hawkular Alerting: Engine .......................... SUCCESS [ 32.529 s]
[INFO] Hawkular Alerting: Actions Implementation .......... SUCCESS [  0.038 s]
[INFO] Hawkular Alerting: Actions Tests ................... SUCCESS [  0.028 s]
[INFO] Hawkular Alerting: Action Plugins .................. SUCCESS [  0.000 s]
[INFO] Hawkular Alerting: Action Email Plugin ............. SUCCESS [ 12.289 s]
[INFO] Hawkular Alerting: Action File Plugin .............. SUCCESS [  1.716 s]
[INFO] Hawkular Alerting: Rest Handlers ................... SUCCESS [  0.567 s]
[INFO] Hawkular Alerting: Integration Tests ............... SUCCESS [03:43 min]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 04:40 min
[INFO] Finished at: 2017-06-28T11:10:20+00:00
[INFO] Final Memory: 45M/442M
[INFO] ------------------------------------------------------------------------

travis_time:end:07b24919:start=1498647937152259380,finish=1498648220788407227,duration=283636147847
[0K
[32;1mThe command "./mvnw -s .travis.maven.settings.xml verify -Pitest-ispn" exited with 0.[0m
```

**Comment by jshaughn** [comment]
Nice time difference in the two runs, ispn running about 30% faster than cassandra.  I expect a lot of that is time that had to be devoted to schema creation for c*.

File: itests/run.sh [hunk 1]
```diff
@@ -7,30 +7,57 @@ if [ ! -d "target/dependency" ]; then
...
+
 JGROUPS_BIND_ADDR="127.0.0.1"
 JAVA_OPTS="$JAVA_OPTS -Xmx64m -Xms64m"
 JAVA_OPTS="$JAVA_OPTS -Djava.net.preferIPv4Stack=true -Djgroups.bind_addr=${JGROUPS_BIND_ADDR}"
 JAVA_OPTS="$JAVA_OPTS -Dmail.smtp.port=2525"
 JAVA_OPTS="$JAVA_OPTS -Dlog4j.configurationFile=src/test/resources/log4j2.xml"
 # JAVA_OPTS="$JAVA_OPTS -agentlib:jdwp=transport=dt_socket,address=8787,server=y,suspend=y"
 
-CLUSTER=$1
+# [lponce] Ispn needs to activate a service from lucene jar and for some reason in standalone mode jar needs to be placed first in classpath
```
Comment (jshaughn) [inline itests/run.sh]: This is weird and will no doubt someday cause some kind of CL issue... Thanks for the comment here.

Comment (lucasponce) [inline itests/run.sh]: Yeah, it is, I spent some time with ispn folks debugging this. It is a thing in how the java service providers work with urls, but this workaround works well.
Note, this is not the final distribution for .next, I will submit a PR with a dist/ folder which will simplify some of this configurations. This one is just for itests.

**Pull request merged**